### PR TITLE
task-6: sqs and sns using added

### PR DESCRIPTION
**Pull request by dmitrii-esin**
Task 6 (AWS SQS/SNS)

All points implemented, but pull-request in refactoring process

- [x] TASK 6.1 Create a lambda function called catalogBatchProcess in the resources section in serverless.yml.
- [x] TASK 6.2 Create a SQS queue, called catalogItemsQueue, configure the SQS to trigger lambda catalogBatchProcess and update the importFileParser lambda
- [x] TASK 6.3 Create an SNS topic createProductTopic and email subscription
- [x] +1 - catalogBatchProcess lambda is covered by unit tests
- [x] +1 - set a Filter Policy for SNS createProductTopic in serverless.yml (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute)

Frontend app link:
   https://d12t0bvcb8pyyn.cloudfront.net/admin/products

import-service:
endpoints:
  GET - https://nr3xtj1l4g.execute-api.eu-west-1.amazonaws.com/dev/import
functions:
  importProductsFile: import-service-dev-importProductsFile
  importFileParser: import-service-dev-importFileParser

product-service:
endpoints:
  GET - https://ie2svy46v9.execute-api.eu-west-1.amazonaws.com/dev/products
  GET - https://ie2svy46v9.execute-api.eu-west-1.amazonaws.com/dev/products/{id}
  POST - https://ie2svy46v9.execute-api.eu-west-1.amazonaws.com/dev/products
  GET - https://ie2svy46v9.execute-api.eu-west-1.amazonaws.com/dev/migrate
functions:
  getAllProducts: product-service-dev-getAllProducts
  getProductById: product-service-dev-getProductById
  createProductHandler: product-service-dev-createProductHandler
  migrateSchema: product-service-dev-migrateSchema
  catalogBatchProcess: product-service-dev-catalogBatchProcess

### CODE REVIEW

File: import-service/src/services/s3-management-service.ts [hunk 1]
```diff
@@ -50,7 +55,43 @@ export class S3ManagementService implements S3ManagementServiceInterface {
...
   }
 
-  async moveFiles(objectKeys: string[]): Promise<unknown[]> {
+  //TODO:!!! types
+  async sendProductsToQueue(products: Product[]): Promise<any> {
+    try {
+      //TODO:!!! types
+      let results: any[] = [];
+
+      products.forEach(async (product) => {
```
Comment (kzanton) [inline import-service/src/services/s3-management-service.ts]: The **forEach** method is not suitable for executing asynchronous code. You cannot use **wait** for that code.

File: import-service/src/services/s3-management-service.ts [hunk 2]
```diff
@@ -50,7 +55,43 @@ export class S3ManagementService implements S3ManagementServiceInterface {
     }
   }
 
-  async moveFiles(objectKeys: string[]): Promise<unknown[]> {
+  //TODO:!!! types
+  async sendProductsToQueue(products: Product[]): Promise<any> {
```
Comment (kzanton) [inline import-service/src/services/s3-management-service.ts]: This method is not suitable for this s3 service

File: import-service/src/services/s3-management-service.ts (hunk 1 above)
Comment (dmitrii-esin) [reply import-service/src/services/s3-management-service.ts]: fixed

File: import-service/src/services/s3-management-service.ts (hunk 2 above)
Comment (dmitrii-esin) [reply import-service/src/services/s3-management-service.ts]: fixed ++

**Pull request merged**